ceptre: move to by-name; add license field; cleanup; update to  0-unstable-2024-8-26

NotAShelf opened the pull request:
## Description of changes

Cleans up derivation for Ceptre, adds license, formats via nixfmt-rfc-style and
updates to the latest untagged commit: 0-unstable-2023-12-12

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See
  [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests)
    (look inside
    [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or
    [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in
    [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or
    [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are
    [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking)
    to the relevant packages
- [x] Tested compilation of all packages that depend on this change using
      `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all
      changes have to be committed, also see
      [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in
      `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md)
  (or backporting
  [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md)
  and
  [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md)
  Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or
        breaking
  - [ ] (Module updates) Added a release notes entry if the change is
        significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS
        module
- [ ] Fits
      [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Frontear reviewed (commented):
Is it possible that rev bumping of this can be automated via a `passthru.updateScript { }`? Would be great so that it doesn't require manual bumping, especially if its a simple drv

Frontear commented:
Result of `nixpkgs-review pr 325113` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ceptre</li>
  </ul>
</details>

NotAShelf commented:
> Is it possible that rev bumping of this can be automated via a `passthru.updateScript { }`? Would be great so that it doesn't require manual bumping, especially if its a simple drv


You mean to undo updating, let  the update bot handle it?

Frontear commented:
> > Is it possible that rev bumping of this can be automated via a `passthru.updateScript { }`? Would be great so that it doesn't require manual bumping, especially if its a simple drv
> 
> You mean to undo updating, let the update bot handle it?

No no, this PR is great and should get merged. My comment meant moreso that in this PR, as part of the cleanup, maybe we can add an updateScript, assuming its possible

NotAShelf commented:
It should be easy to do, I'll handle that tomorrow.

SuperSandro2000 commented:
Please rebase

NotAShelf commented:
I had missed that, done.

NotAShelf commented:
Formatting is fixed too now.

Pandapip1 commented:
Merge conflict due to https://github.com/NixOS/nixpkgs/pull/354531.

NotAShelf commented:
That was... unexpected. I'll try to fix it tomorrow.

NotAShelf committed:
ceptre: format via nixfmt

NotAShelf committed:
ceptre: bump to 0-unstable-2024-8-26

NotAShelf committed:
ceptre: cleanup

Add myself as a maintainer so that I'm made aware of future updates, and get rid of nested `with lib`

NotAShelf commented:
Okay that was faster than I thought. @Pandapip1 I hope this is satisfactory?

Pandapip1 commented:
![image](https://github.com/user-attachments/assets/ab9ad4e1-14b7-40e5-87f5-e1ab9a08f4e4)

LGTM

SuperSandro2000 commented on pkgs/by-name/ce/ceptre/package.nix:
```diff
@@ -1,29 +1,39 @@
+      NotAShelf
+      pSub
+    ];
+    platforms = lib.platforms.unix;
```
```suggestion
    platforms = lib.platforms.unix;
    license = lib.licenses.unfree;
```

NotAShelf replied:
Looks like it got messed up during the rebase, fixed.

niklaskorz commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 325113`

niklaskorz commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 325113`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ceptre</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2182
